src/tezos_interop/tezos_js_bridge.js

```diff
@@ -124,13 +139,56 @@ const onRequest = async (request) => {
   return { status, hash };
 };
 
+/** @param {StorageRequest} request */
+const onStorageRequest = async (request) => {
+  const { rpc_node, confirmation, destination } = request;
+  const client = new RpcClient(rpc_node);
+  const Tezos = new TezosToolkit(rpc_node);
+  Tezos.setProvider({ config });
+
+  const block = await client.getBlock(); // fetches the head
+  const contract = await client.getContract(destination, {
+    block: block.hash,
+  });
+  const storage = contract.script.storage;
```

Comment:
Why extracting it so long before needing it? I expected to find some transformation on it. I think this line should be right before the `return` one.

Reply:
I just kept the code in the same order, this is quite fragile and I didn't want to break it.